app-cpp/app355.cpp

```diff
@@ -68,38 +69,234 @@ class OrbitalData
+                threadVector.push_back(std::thread(&SatOrbit::OnSortOrbitalVectorMulti, this, orbitBegin, orbitEnd));
+
+                IteratorPairVector::value_type slice{ orbitBegin, orbitEnd };
+                subListVector.push_back(slice);
+            }
+            
+            // Join the threads together
+            // wait for the threads to finish before returning
+            std::for_each(threadVector.begin(), threadVector.end(), [](std::thread& inThread)
+            {
+                inThread.join();
+            });
+        }
+
+        // Merge the sorted sublists
+        // Use multithreading by merging multilple sublists at a time
+        // Join the threads before merging the set of larger sublists
+        // Loop until there is only one list of completely sorted OrbitalData
+        while (subListVector.size() > 1)
+        {
+            // Merge sort the sublists 2 at a time
+            std::vector<std::thread> threadVector{};
+            IteratorPairVector newSubListVector{};
+
+            for (std::size_t i = 1; i < subListVector.size(); i += 2)
+            {
+                // Get first 2 sublists to merge
+                auto& [begin, mid] = subListVector[i-1];
+                auto& [mid2, end] = subListVector[i];
+                // Mids must be sequential in memory
+                assert(mid == mid2);
+
+                // Start a thread to merge the 2 sublists
+                threadVector.push_back(std::thread(&SatOrbit::OnSortMergeVector, this, begin, mid, end));
+                // Create a new sublist element from the 2 merged sublists onto a new list
+                newSubListVector.push_back(std::make_tuple(begin, end));
+            }
+            // Wait for the threads to finish their merge operation
+            std::for_each(threadVector.begin(), threadVector.end(), [](std::thread& inThread)
+            {
+                inThread.join();
+            });
+
+            // If the number of sublists is odd, it would be missed in the loop above; add it to the new list
+            const bool isOdd = ((subListVector.size() & 1) == 1);
+            if (isOdd)
+            {
+                newSubListVector.push_back(subListVector.back());
+            }
+
+            // Update subListVector with the new merged results
+            subListVector = std::move(newSubListVector);
+        }
+    }
+}
+
+std::vector<std::vector<OrbitalData>> SatOrbit::CreateTrains(const std::vector<OrbitalData>& orbitalVector)
+{
+    return OnCreateTrains(orbitalVector);
+}
+
+void SatOrbit::PrintTrains(const std::vector<std::vector<OrbitalData>>& trainVector)
+{
+    OnPrintTrains(trainVector);
+}
+
+// NVI Implementation: Private Virtuals
+
+std::vector<sat355::TLE> SatOrbit::OnReadFromFile(int argc, char* argv[])
 {
     if (argc < 2) 
     {
         std::cout << "Please provide a file path." << std::endl;
-        //return 1;
```

Comment:
`return 1;`... meaning "an error occurred".
in `C` we handle this by "returning an error code".
so every `C` routine has a return value that is the error code.

in `C++` we handle this via: "throwing an exception".
I would suggest throwing an exception defined by `std::filesystem`.
Look it up and find an exception to throw